### PR TITLE
chore: Changes for release build job

### PR DESCRIPTION
**Pull request by ddixit14**

### CODE REVIEW

**Comment by ddixit14** [comment]
> Why remove the release profile? It prevents samples and test project from getting released.
> 
> https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/blob/master/pom.xml#L304-L306

Sorry, it was by mistake. I actually intended to include the profile in the promote release part of the script, since the modules being processed in both the staging and releasing scripts were different.

**Review by elefeint — CHANGES_REQUESTED** [review]
Do you need to specify the staging repository? In the old process, stage.sh sent all releasable artifacts to Sonatype, and then promote.sh triggered the publishing of those artifacts. This was all tied together by the "staging repository ID" -- the thing that you'd find in the staging logs and provide to the release/promote job.

The current [promote.sh](https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/blob/fabbc81e3924b1f12658760c5bf3e25eb0ecba4d/.kokoro/promote.sh#L39) passes `-DstagingRepositoryId=${STAGING_REPOSITORY_ID}` to let sonatype know which pre-staged directory to publish. What is the new process meant to do -- release in one step, or stage-and-release like before just with a single script?

File: .kokoro/stage.sh [hunk 1]
```diff
@@ -46,9 +47,10 @@ create_settings_xml_file $MAVEN_SETTINGS_FILE
 # promote release
 if [[ -n "${AUTORELEASE_PR}" ]]
 then
-  mvn nexus-staging:release -B \
+  ./mvnw nexus-staging:release -B \
     -DperformRelease=true \
-    --settings=settings.xml
+    --settings=settings.xml \
+    -P release
```
Comment (suztomo) [inline .kokoro/stage.sh:53]: The missing problem. The 2nd command was trying to release the integration test module.

**Review by suztomo — APPROVED** [review]
Approving. Don't forget to answer Elena's question.

**Comment by elefeint** [comment]
We've looked at it offline, and it looks like `mvn deploy` stages it first locally, and then remotely at Sonatype, generating the staging repository ID along the way. Then `mvn nexus-release` takes the already-staged files and the (cached?) staging repository ID and allows promotion of staged artifacts to released ones.

**Comment by ddixit14** [comment]
@elefeint can you please approve this so I can merge it in?

File: pom.xml [hunk 2]
```diff
@@ -175,7 +175,7 @@
           <version>1.6.13</version>
           <extensions>true</extensions>
           <configuration>
-            <serverId>ossrh</serverId>
+            <serverId>sonatype-nexus-staging</serverId>
```
Comment (elefeint) [inline pom.xml:178]: I'd keep `ossrh` in all three instances, unless there is a reason to change repository ID.

**Review by elefeint — CHANGES_REQUESTED** [review]

**Review by elefeint — APPROVED** [review]

**Pull request merged**